controllers/templatesync/template_sync.go

```diff
@@ -121,6 +126,19 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, nil
 	}
 
+	allDeps := make(map[depclient.ObjectIdentifier]bool)
```

Comment:
Optional - it would be a bit cleaner to use something like https://github.com/deckarep/golang-set since we don't use the bool value, but up to you if you don't want to import another library

Reply:
I don't know if there's enough advantage to using another library for this, I think it would only clean up the conversion to a slice? If we had more set-specific operations like intersections or joins, then I'd be happy to use the library.